### ISSUE TITLE
Management REST API error messages

### ISSUE THREAD

**Issue by Nemikolh** [GreenPix/lycan#20]
The management REST API does not produce valid json when there's an error which would make writing a UI on top of it rather annoying. I think we could return instead something like:

```
{
    "type": "error",
    "message": "ERROR: invalid id 0: Invalid length; expecting 32 or 36 chars, found 1"
}
```

Similarly if an API is always expected to return a response in case of success we could have something like:

```
{
    "type": "success",
    "message": "Monster spawned with id '234234'"
}
```

**Comment by ghost** [comment]
The management API specification is still a WIP (see https://github.com/GreenPix/meta/pull/11) so indeed I didn't polish the implementation, particularly the error cases (where I just put a HTTP error code).

As you know, I have never made a real UI, so any help on what would make it easy to write one is most welcome.

**Comment by lthms** [comment]
The HTTP code of the response already gives the type of the response body, rather a success (2xx) or an error (basically all the other ones). Therefore, I think the `type` field is useless.

That being said, a JSON formatted error would be nice. For instance:

``` json
{
    "code": 1, /* where 1 is specified to be "ID not found error" */
    "data": { "provided": "10" }
}
```

And we can imagine many sort of error code (per REST route? for the all application?), with very different data attached to it.

**Comment by Nemikolh** [comment]
True! Using the http code here is more appropriate. However, I don't like
the "code" idea at all neither having different format of error all over
the place. I find the "message" field strictly superior for the following
reasons :
- The precise error reason can be easily expressed by the server with a
  string than by a generic error message mechanism that might not cover all
  the needs before becoming a beast.
- It does not couples the consumer code with the Lycan error one as in your
  system both needs to be updated when an error shape change.
- It means less work for both Lycan and the consumer of the API (consumer
  is a single line in js: `error.message`.
- The consumer can handle all errors uniformly.

Now, we could also have both the message field as well as the structured
part. Structured errors are strictly superior for advanced error reporting
but I don't think it is appropriate here. A message seems to be more than
enough to me.

**Comment by lthms** [comment]
If we only stick with an error message, then I do not see the point of wrapping it into a json object. Indeed, a String is, AFAIK, a valid json input. Of course, wrapping it allows us to change our mind in the future with a far less important cost than if you do not.

**Comment by Nemikolh** [comment]
This is true as long as the returned data is a quoted escaped string (we
tend to forget about that tiny detail). ;-)

I agree with you, returning a json string would be enough. Your later point
on extensibility is a very good point!

To be honest, I'm fine with the final decision as long as the output is a
json that gives access to an easy to use error message. :-)
